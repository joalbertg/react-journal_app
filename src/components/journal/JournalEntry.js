import React from 'react';
import moment from 'moment';
import { useDispatch } from 'react-redux';

import { activeNote } from '../../actions';

export const JournalEntry = ({ note }) => {
  const { id, title, body, date, url } = note;
  const noteDate = moment(date);
  const dispatch = useDispatch();

  const handleEntryClick = () => {
    dispatch(activeNote(id, note));
  }

  return(
    <div
      onClick={handleEntryClick}
      className='journal__entry pointer'
    >
    { url &&
        <div
          className='journal__entry-picture'
          style={{
            backgroundSize: 'cover',
            backgroundImage: `url(${url})`
          }}
        >
        </div>
    }
      <div className='journal__entry-body'>
        <p className='journal__entry-title'>
          {title}
        </p>
        <p className='journal__entry-content'>
          {body}
        </p>
      </div>
      <div className='journal__entry-date-box'>
        <span>{noteDate.format('dddd')}</span>
        <h4>{noteDate.format('Do')}</h4>
      </div>
    </div>
  );
}

