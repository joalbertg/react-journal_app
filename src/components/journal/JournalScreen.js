import React from 'react';

import { Sidebar } from './Sidebar';
//import { NothingSelected } from './NothingSelected';
import NoteScreen from '../notes';

const JournalScreen = () => {
  return(
    <div className='journal__main-content'>
      <Sidebar />
      <main>
        {/*<NothingSelected />*/}
        <NoteScreen />
      </main>
    </div>
  );
}

export default JournalScreen;

