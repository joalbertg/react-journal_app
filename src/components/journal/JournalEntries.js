import React from 'react';
import { useSelector } from 'react-redux';

import { JournalEntry } from './JournalEntry';

export const JournalEntries = () => {
  const { notes } = useSelector(state => state.notes);
  const indexes = Object.keys(notes);

  return(
    <div className='journal__entries'>
      {
        indexes.map(index => (
          <JournalEntry
            key={index}
            note={notes[index]} />
        ))
      }
    </div>
  );
}

