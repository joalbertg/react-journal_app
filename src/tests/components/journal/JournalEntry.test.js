import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';

import { JournalEntry } from '~components/journal/JournalEntry';
import { activeNote } from '~actions';

//jest.mock('~actions', () => ({
//  activeNote: jest.fn()
//}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const store = mockStore({});

store.dispatch = jest.fn();

describe('Tests JournalEntry component', () => {
  const note = {
    id: 'ABCD1234',
    title: 'Testing title',
    body: 'Body title',
    date: '2020-07-07T00:49:11.159Z'
  };

  const wrapper = mount(
    <Provider store={store}>
      <JournalEntry note={note} />
    </Provider>
  );

  test('Should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should call activeNote', () => {
    wrapper.find('.journal__entry').prop('onClick')();

    //expect(activeNote).toHaveBeenCalled();
    //expect(activeNote).toHaveBeenCalledTimes(1);
    //expect(activeNote).toHaveBeenCalledWith(note.id, note);

    // another way to avaluate without mocking
    expect(store.dispatch).toHaveBeenCalledWith(activeNote(note.id, note));
  });
});

