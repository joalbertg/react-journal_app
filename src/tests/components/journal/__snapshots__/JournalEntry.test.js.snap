// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests JournalEntry component Should display correctly 1`] = `
<div
  className="journal__entry pointer animate__animated animate__fadeIn animate__faster"
  onClick={[Function]}
>
  <div
    className="journal__entry-body"
  >
    <p
      className="journal__entry-title"
    >
      Testing title
    </p>
    <p
      className="journal__entry-content"
    >
      Body title
    </p>
  </div>
  <div
    className="journal__entry-date-box"
  >
    <span>
      Monday
    </span>
    <h4>
      6th
    </h4>
  </div>
</div>
`;
